db/version_set.cc

```diff
@@ -1978,7 +1979,14 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
     *key_exists = true;
   }
 
-  PinnedIteratorsManager pinned_iters_mgr;
+  PinnedIteratorsManager local_pinned_iters_mgr;
```

Comment:
Do we still need to allocate it here if in the common path, there is one passed in?

Reply:
Yep - still working around it but couldn't think of another solution to bypass the allocation other than (void) it later, )(actually not even sure if voidING it actually frees the memory earlier).

The issue I had is I tried to some continually allocation like 
```
if (use_local_pinned_iters_mgr) {
PinnedIteratorsManager local_pinned_iters_mgr;
pinned_iters_mgr = & local_pinned_iters_mgr;
```

but then local_pinned_iters_mgr will local to the conditional branch scope. 

Maybe I'm just not aware of a general coding pattern to this kind of issue - any thought?

Reply:
Can we always force the caller to allocate one?

Reply:
Good question - I was thinking of a minimum code change so didn't touch any of the callers. But yeah we could do that to all the callers (6 of them) and make sure the passed-in `PinnedIteratorsManager ` life-time is well taken cared.  Will update in the next draft.